test/runtime/document_test.cc

```diff
@@ -72,19 +72,6 @@ describe("Document", [&]() {
         "(array (true) (false))");
     });
 
-    it("handles truncated UTF16 data", [&]() {
-      char *content = reinterpret_cast<char*>(malloc(1));
-
-      spy_input->content = string((const char *)content, 1);
-      spy_input->encoding = TSInputEncodingUTF16;
-
-      ts_document_set_input(document, spy_input->input());
-      ts_document_invalidate(document);
-      ts_document_parse(document);
-
-      free(content);
-    });
-
```

Comment:
You're totally right, much better to go through `ts_document_parse`. I thought it was some magic in `std::string` but I tracked it down to [the spy buffer logic](https://github.com/tree-sitter/tree-sitter/blob/20b664969e8ec2a5aa94754f41975e8792522fa4/test/helpers/spy_input.cc#L64) which is helpfully padding allocations and initialising memory, which totally defeats valgrind 😄